pkg/kvstore/etcd.go

```diff
@@ -238,39 +237,35 @@ func getEPVersion(c client.Maintenance, etcdEP string, timeout time.Duration) (*
 }
 
 // checkMinVersion checks the minimal version running on etcd cluster. If the
-// minimal version running doesn't meet cilium minimal requirements, returns
-// an error.
-func (e *etcdClient) checkMinVersion(timeout time.Duration) error {
+// minimal version running doesn't meet cilium minimal requirements, returns an
+// error.
+func (e *etcdClient) checkMinVersion() {
 	eps := e.client.Endpoints()
-	var errors bool
+
 	for _, ep := range eps {
-		v, err := getEPVersion(e.client.Maintenance, ep, timeout)
+		v, err := getEPVersion(e.client.Maintenance, ep, versionCheckTimeout)
 		if err != nil {
-			log.WithError(err).Debug("Unable to check etcd min version")
-			log.WithError(err).WithField(fieldEtcdEndpoint, ep).Warn("Checking version of etcd endpoint")
-			errors = true
+			log.WithError(err).WithField(fieldEtcdEndpoint, ep).
+				Warn("Unable to verify version of etcd endpoint")
```

Comment:
Do we want to continue onward if we are unable to verify the version? If we are unable to get the version and the version is < 3.1.0, what behavior would be observed in Cilium?

Reply:
I thought of this too, but the go routine is a fatal error, so even if it takes a while to check the version, we will eventually cause a fatal error if there is a mismatch.

Reply:
If we fail there, we would see exactly the behavior we are showing now, if any of the etcd endpoints become unavailable, all cilium agents would die on startup or session renewal.

Reply:
We can potentially become smarter in understanding the reason why we can't verify the version but it seems non trivial for now

Reply:
Sounds good to me, thanks!